### DISCUSSION TITLE
Remove multiple deployments

### DISCUSSION THREAD

**Discussion by caenguidanos** [vercel/vercel#5579] [General] [answered]
Hi Vercels,

How you remove multiple deployments?

The unique idea i have is something like this:

#### 1. List deployments with CLI

`vercel ls <project>`

#### 2. Copy result to script:

```js
const { execSync } = require('child_process');

const s = `
deployment-1.vercel.app,
deployment-2.vercel.app,
deployment-3.vercel.app,
deployment-4.vercel.app,
...
`;

function removeDeployments() {
  const dps = s.split(',').map(v => v.trim());

  for (const i of dps) {
    execSync(`vercel rm -y ${i}`);
  }
}

removeDeployments()
```

**Comment by mcsdevv** [comment]
Hello @cenguidanos, there should be no reason to delete deployments. However, the easiest way to do so is with the `vc remove --safe` command, you can find more information [here](https://vercel.com/docs/cli#commands/remove/unique-options/safe).

**Reply by benalexb** [reply]
Hi @mcsdevv, I'm curious about _"there should be no reason to delete deployments"_. Does that mean I can have unlimited deployments? Wouldn't that be resource intensive and expensive? Honest question. Currently, it is very time consuming for me to go back and remove old deployments, and it would be nice to keep them (github PRs have a reference to the build status of that revision, which is naturally broken once the deployment is deleted). I have only been removing them for fear of having a limit, or getting charged.

**Reply by mcsdevv** [reply]
Hi @benalexb, absolutely! There is currently nothing within our limits in relation to the amount of both projects and deployments a customer can have. If this was to ever change it would be something I'd absolutely expect some very clear messaging on and much greater tooling for the management of both items, the only exception perhaps being if this was an effort to curb platform abuse.

**Reply by benzler** [reply]
Hey @mcsdevv I'm just curious, isn't it a security issue if old deployments with outdated packages are still running?

**Reply by mcsdevv** [reply]
Hey @benzler, sorry for the delay here!

While it's highly unlikely these deployments would be found in the wild (unless widely shared), there'a certainly a possibility. As such, we're working on a couple of solutions that help address this problem entirely. As a customer, you shouldn't have to worry about deleting deployments, but you also shouldn't have to worry about security.

**Reply by benzler** [reply]
Thanks for your reply. Then I'm looking forward to these solutions :)

**Reply by mnieber** [reply]
Hi @mcsdevv I came to this thread because I also wondered if previous deployments are kept alive. In general, I try to be resource-aware to do my part in reducing the burden on the planet. Do you think that for "green reasons" I should delete old deployments, or will this only have a marginal effect?

**Reply by mcsdevv** [reply]
Hey @mnieber, while the effect would be marginal, it's still an effect!

**Comment by ZoranRavic** [comment]
`vercel rm` can accept multiple url's

so this:

```js
function removeDeployments() {
  const dps = s.split(',').map(v => v.trim());

  for (const i of dps) {
    execSync(`vercel rm -y ${i}`);
  }
}
```

can be shortened to:

```js
function removeDeployments() {
  const dps = s.split(',').map(v => v.trim());

  execSync(`vercel rm -y ${dps.join(' ')}`);
}
```

**Comment by leerob** [comment, marked as answer]
You can now configure the retention period for deployments through the dashboard and CLI.

For example, canceled and errored deployments might be set to 30 days retention, while production deployments might be set to 1 year. Recently deleted deployments are shown in your project settings and can be instantly restored within 30 days of deletion.

Learn more in our [documentation](https://vercel.com/docs/security/deployment-retention).
